rust/tw_hash/src/ffi.rs

```diff
@@ -31,7 +31,9 @@ pub unsafe extern "C" fn blake2_b(
     hash_size: usize,
 ) -> CByteArray {
     let input = std::slice::from_raw_parts(input, input_len);
-    blake2::blake2_b(input, hash_size).into()
+    blake2::blake2_b(input, hash_size)
+        .unwrap_or_default()
```

Comment:
Do we generally want to simply return an empty array on error? I guess there are multiple design decisions on how to implement error handling, up for debate.

Reply:
Thank you for the note!
I've integrated `CResult` wrapper over different type like `CByteArray`, `CStr` etc.
https://github.com/trustwallet/wallet-core/blob/1823ad8cc2c52645336909734e212e4605a16596/rust/tw_encoding/src/ffi.rs#L96-L109

I agree with you that we should return `CByteArrayResult` here as well.

Reply:
Done